torch/ao/quantization/pt2e/_numeric_debugger.py

```diff
@@ -71,6 +71,45 @@ def _assign_debug_handle(node: torch.fx.Node) -> None:
     bfs_trace_with_node_process(ep, _assign_debug_handle)
 
 
+def _detach(x):
+    if isinstance(x, torch.Tensor):
+        detached = x.detach()
+    elif isinstance(x, (list, tuple)):
+        detached = type(x)([_detach(e) for e in x])
+    elif isinstance(x, dict):
+        detached = {k: _detach(e) for k, e in x.items()}
+    else:
+        detached = x
+    return detached
+
+def _tensor_shape_equals(x, y):
```

Comment:
For these two functions, see if you can reuse `map_aggregate` from fx.node. It already handles all the tuple/dict branching stuff.
You can give it a callable for Tensor to do the detach or replace a Tensor with its shape for comparison

Reply:
I can use map_aggregate for `_detach`, but others does not help that much I think

Reply:
actually I'll revert this since it introduced a lot of typing errors..